conversion/conversion.go

```diff
@@ -1280,6 +1300,67 @@ func ProcessDump(driver string, conv *internal.Conv, r *internal.Reader) error {
 	}
 }
 
+func GetInfoSchemaFromCloudSQL(sourceProfile profiles.SourceProfile, targetProfile profiles.TargetProfile) (common.InfoSchema, error) {
+	driver := sourceProfile.Driver
+	switch driver {
+	case constants.MYSQL:
+		d, err := cloudsqlconn.NewDialer(context.Background(), cloudsqlconn.WithIAMAuthN())
+        if err != nil {
+                return nil, fmt.Errorf("cloudsqlconn.NewDialer: %w", err)
+        }
+        var opts []cloudsqlconn.DialOption
+		instanceName := fmt.Sprintf("%s:%s:%s", sourceProfile.ConnCloudSQL.Mysql.Project, sourceProfile.ConnCloudSQL.Mysql.Region, sourceProfile.ConnCloudSQL.Mysql.InstanceName)
+        mysqldriver.RegisterDialContext("cloudsqlconn",
+                func(ctx context.Context, addr string) (net.Conn, error) {
+                        return d.Dial(ctx, instanceName, opts...)
+                })
+
+        dbURI := fmt.Sprintf("%s:empty@cloudsqlconn(localhost:3306)/%s?parseTime=true",
```

Comment:
is the port always going to be 3306 ? Can this be fetched from the instance ? 
Is it called out in the documentation ?

Reply:
This port cannot be changed in CloudSQL and is always 3306 for MySQL -
https://stackoverflow.com/questions/32354991/change-tcp-port3306-for-google-cloud-sql-instance